NETOBSERV-1322: ACM & netobserv metrics (#61)
netobserv/documents: acm.md

> ## Setup ACM with NetObserv metrics

Author: for reviewers: this file is just a recipe for internal purpose, not the blog post; for the blog, look at `blogs/acm/leverage-metrics-in-acm.md`

cf also [blog post](./blogs/acm/leverage-metrics-in-acm.md).

This is more a quick guide for the development teams.

Quick guide:

> 1. Create 2 clusters (or more)
> 2. Choose one for being the main one / hub: install ACM operator on it; Create a default MultiClusterHub
> 3. In console top bar, select "all cluster" then start procedure to import an existing cluster

Reviewer: ```suggestion
1. Create 2 clusters (or more).
2. Choose one cluster as the main one or hub, and install the ACM operator on it.
3. Create a default MultiClusterHub.
4. In the console top bar, select "all cluster" then start the procedure to import an existing cluster.
```

Reviewer: A few comments about these steps:

- I think "create a default MultiClusterHub" should be its own step, or a nested step rather than a continuation of the step 3 with a semicolon. 
- Is this a MultiClusterHub custom resource or Operator? I think specifying would be good. 
- I'm wondering, should ACM be spelled out? Or is it an approved acronym that all readers would be familiar with? If its new, I would suggest spelling it out and putting ACM in parentheses for this first mention, then elsewhere you can just use ACM. 
- Periods are needed at the end of these steps.
- I also added some "the".

> On each cluster:

Reviewer: ```suggestion
On each cluster, perform the following steps:
```

> 1. Install netobserv downstream (user workload prometheus won't work)

Reviewer: ```suggestion
1. Install network observability operator downstream (user workload Prometheus won't work).
```

> 2. Create a FlowCollector, with these metrics enabled (`spec.processor.metrics.includeList`) :

Reviewer: ```suggestion
2. Create a FlowCollector custom resource, with the following metrics enabled (`spec.processor.metrics.includeList`) :
```

```yaml
      includeList:
        - namespace_flows_total
        - node_ingress_bytes_total
        - workload_ingress_bytes_total
        - workload_egress_bytes_total
        - workload_egress_packets_total
        - workload_ingress_packets_total
```

cf steps at https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/observability/observing-environments-intro#enabling-observability :

```bash
oc create namespace open-cluster-management-observability
DOCKER_CONFIG_JSON=`oc extract secret/pull-secret -n openshift-config --to=-`
oc create secret generic multiclusterhub-operator-pull-secret \
    -n open-cluster-management-observability \
    --from-literal=.dockerconfigjson="$DOCKER_CONFIG_JSON" \
    --type=kubernetes.io/dockerconfigjson
```

Setup S3, Thanos Secret and ACM observability:

```bash
./examples/ACM/thanos-s3.sh yourname-thanos us-east-2
oc apply -f examples/ACM/acm-observability.yaml
oc get pods -n open-cluster-management-observability -w
oc apply -f examples/ACM/netobserv-metrics.yaml 
```

> To debug the above config, check logs here:

Reviewer: ```suggestion
To debug the above config, check the logs here:
```

```bash
oc logs -n open-cluster-management-addon-observability -l component=metrics-collector
```

> Deploying dashboards:

Reviewer: ```suggestion
Deploy the dashboards:
```
